src/main/java/seedu/address/logic/commands/FindAppointmentCommand.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DOCTOR;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIMESLOT_START;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all appointments in the current Appointment Schedule
+ * based on matching tags.
+ * Keyword matching is case insensitive.
+ */
+public class FindAppointmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "find-appt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " Finds all appointments from the appointment schedule "
+            + "in which patients, doctors, timeslots, or tags contain any of "
+            + "the specified keywords (case-insensitive) and "
+            + "displays them as an indexed list.\n"
+            + "Parameters: (Any one or more)"
+            + PREFIX_NAME + "NAME "
+            + PREFIX_DOCTOR + "DOCTOR "
+            + PREFIX_TIMESLOT_START + "TIMESLOT START "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD
+            + PREFIX_TAG + "brainDamage";
+
+    private final AppointmentContainsKeywordsPredicate predicate;
+
+    public FindAppointmentCommand(AppointmentContainsKeywordsPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredAppointmentList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, model.getFilteredAppointmentList()
+                        .size()));
```

Comment:
is there a need to 8 spaces then 8 spaces again? im not too sure about the coding convention